pytorch_lightning/trainer/trainer.py

```diff
@@ -2287,6 +2289,47 @@ def terminate_on_nan(self, val: bool) -> None:
         )
         self._terminate_on_nan = val  # : 212
 
+    def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[str, dict]:
+        """Called when saving a model checkpoint.
+
+        Calls every callback's `on_save_checkpoint` hook.
+        """
+        callback_states = {}
+        for callback in self.callbacks:
+            state = callback.on_save_checkpoint(self, self.lightning_module, checkpoint)
+            if state:
+                callback_states[callback.state_key] = state
+        return callback_states
+
+    def _call_callbacks_on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """Called when loading a model checkpoint.
+
+        Calls every callback's `on_load_checkpoint` hook.
+        """
+        # Todo: the `callback_states` are dropped with TPUSpawn as they
+        # can't be saved using `xm.save`
+        # https://github.com/pytorch/xla/issues/2773
```

Comment:
@kaushikb11 is this comment here relevant? i don't see any special handling for this here. 

I see it's also referenced here, where the callback states are explicitly not saved: https://github.com/PyTorchLightning/pytorch-lightning/blob/08e661ff7266f298a6e904fa04931698e20b2b89/pytorch_lightning/plugins/io/xla_plugin.py#L43-L47